### DISCUSSION TITLE
Automatic redirects to wayback machine (Internet archive)

### DISCUSSION THREAD

**Discussion by Aravind-pro** [cefsharp/CefSharp#3672] [General]
![image](https://user-images.githubusercontent.com/78254185/124414889-31b63600-dd71-11eb-893b-b57fa994b70d.png)
I had made a web browser using Cefsharp . In my web browser, there is a bug which redirects to wayback machine while browsing. It always redirects automaticaly  .Please help me to fix this issue . I am using Cefsharp version 86 for winforms

**Comment by amaitland** [comment]
You'll need to provide a code example that reproduces the problem. 

In future please use https://github.com/cefsharp/CefSharp/discussions to ask questions rather than opening an issue.

**Reply by Aravind-pro** [reply]
. Next time onwards I will use discussions tab.Can you please help me ?

**Reply by amaitland** [reply]
Please provide code example as requested. 

I cannot help without seeing your code.

**Reply by amaitland** [reply]
Where is the code that loads the internet archive?

**Reply by amaitland** [reply]
Use a code sharing service like https://gist.github.com/

Put all your code in a single gist so it is easy to searched.

You need to make it as easy as possible to review your code.

**Reply by Aravind-pro** [reply]
https://gist.github.com/Aravind-pro/2d5575a631a0fae82bd64ae0cae34ca3

**Reply by amaitland** [reply]
There doesn't appear to be a single CefSharp reference in the code you provided. 

All I'm interested in is the CefSharp specific pieces of code, specifically where you are redirecting to internet archive.

**Reply by amaitland** [reply]
Where is the code that redirects to internet archive?

**Reply by Aravind-pro** [reply]
I don't know which code redirects to internet archive and why . I only used these codes provided above. I first tried changing `WebGL = `false`` But the problem persists . Then I thought of deleting this code `CefSharpSettings.WcfEnabled = false;`` but it doesn't worked

**Reply by Aravind-pro** [reply]
This is the error . I am providing video so that you can easily understand the problem
https://user-images.githubusercontent.com/78254185/124457237-33035500-dda9-11eb-9109-bf6a44c3a2be.mp4

**Reply by amaitland** [reply]
> I don't know which code redirects to internet archive and why . I only used these codes provided above. 

Where is your RequestHandler code?

> browser.RequestHandler = rHandler;

Comment out this line for testing purposes.

**Reply by Aravind-pro** [reply]
Thanks you so much . I found the mistake in Request handler and corrected it . I deleted the code

**Reply by Aravind-pro** [reply]
`
			// if NOT FOUND
			if (code == 404) {

				if (!request.Url.IsURLLocalhost()) {

					// redirect to web archive to try and find older version
					frame.LoadUrl("http://web.archive.org/web/*/" + request.Url);

				}
				else {

					// show offline "file not found" page
					frame.LoadUrl(MainForm.FileNotFoundURL + "?path=" + request.Url.EncodeURL());
				}

			}`

**Reply by Aravind-pro** [reply]
I corrected this mistake but there is a problem.Many links aren't clickable after clearing the code Can you please help me here`frame.LoadUrl("http://web.archive.org/web/*/" + request.Url);
`

**Reply by amaitland** [reply]
How does the url behave if you load it directly in Chrome?

**Reply by Aravind-pro** [reply]
Everything is fine in Chrome . I will give an example.  When I check mail in Netcliff(My browser) , The links I recieve from anyone in mail is unclickable. It  works only after right clicking on the link and click open in new tab button

**Reply by Aravind-pro** [reply]
Will upgrading to Cefsharp version 90 , fix this issue

**Reply by amaitland** [reply]
Version 91 is the current supported version. You can test using https://github.com/cefsharp/CefSharp.MinimalExample

As I've never seen your application and I don't have access to the source the examples you give don't mean anything to me.

**Reply by Aravind-pro** [reply]
``			if (!request.Url.IsURLLocalhost()) {

				// redirect to web archive to try and find older version
				frame.LoadUrl("http://web.archive.org/web/*/" + request.Url);

			}
			else {

				// show offline "file not found" page
**Can you please help me in this code**

**Reply by amaitland** [reply]
Is your project open source? 

Try loading the url directly in the minimal example.

**Reply by amaitland** [reply]
Please take the time to read https://stackoverflow.com/help/minimal-reproducible-example

**Reply by amaitland** [reply]
You can open DevTools and check for errors, look at the request/response. 

https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#javascript-debugging

**Reply by Aravind-pro** [reply]
I fixed it!. I change the user agent code `public static string UserAgent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.164 Safari/537.36";`
To Chrome version 91 .
